lib/server.js

```diff
@@ -826,12 +825,32 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
 
         if (arg) {
             if (arg instanceof Error) {
-                errName = arg.name.replace(/Error$/, '');
-                log.trace({err: arg, errName: errName}, 'next(err=%s)',
-                    (arg.name || 'Error'));
 
+                var errName = arg.name.replace(/Error$/, '');
+                log.trace({
+                    err: arg,
+                    errName: errName
+                }, 'next(err=%s)', (arg.name || 'Error'));
+
+                // always attempt to use the most specific error listener
+                // possible. fall back on generic 'error' listener if we can't
+                // find one for the error we got.
+                var hasErrListeners = false;
+                var errEvtName;
+
+                // if we have listeners for the specific error
                 if (self.listeners(errName).length > 0) {
-                    self.emit(errName, req, res, arg, once(function () {
+                    hasErrListeners = true;
+                    errEvtName = errName;
+                }
+                // or if we have a generic error listener
+                else if (self.listeners('error').length > 0) {
+                    hasErrListeners = true;
+                    errEvtName = 'error';
```

Comment:
I went with 'nextError', but happy to change it to 'restifyError'. I thought it might reduce the number of questions we get around the difference between InternalError vs InternalServerError vs uncaughtException, but maybe that's a docs issue.